docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc

```diff
@@ -386,6 +386,186 @@ It is possible to use multiple expressions in the role definition.
+    @PermissionsAllowed("update")
+    @POST
+    public String createOrUpdate() {
+        return "modified";
+    }
+
+    @PermissionsAllowed({"see:detail", "see:all", "read"}) <2>
+    @GET
+    @Path("/id/{id}")
+    public String getItem(String id) {
+        return "item-detail-" + id;
+    }
+
+    @PermissionsAllowed(value = "permission-1", permission = CustomPermission.class) <3>
+    @GET
+    public Collection<String> list(@QueryParam("query-options") String queryOptions) {
+        // your business logic comes here
+        return Collections.emptySet();
+    }
+
+    public static class CustomPermission extends Permission {
+
+        public CustomPermission(String name) {
+            super(name);
+        }
+
+        @Override
+        public boolean implies(Permission permission) {
+            var event = Arc.container().instance(RoutingContext.class).get(); <4>
+            return "show-first-page".equals(event.request().params().get("query-options"));
+        }
+
+        ...
+    }
+}
+----
+<1> Resource method `createOrUpdate` is only accessible by user with both `create` and `update` permissions.
+The `@PermissionsAllowed` is repeatable and all annotation instances are conjuncts.
+<2> User is granted access to `getItem` if he possess either `read` permission or `see` permission and one of actions (`all`, `detail`).
+Relation between permissions specified through `PermissionsAllowed#value` is disjunctive.
+<3> You can use any `java.security.Permission` implementation of your choice.
+By default, string-based permission is performed by the `io.quarkus.security.StringPermission`.
+<4> Permissions are not beans, therefore only way to obtain bean instances is programmatically via the `Arc.container()`.
+
+CAUTION: If you plan to use the `@PermissionsAllowed` on the IO thread, review the information in xref:security-proactive-authentication-concept.adoc[Proactive Authentication].
+NOTE: The `@PermissionsAllowed` is not repeatable on class-level due to limitations of Quarkus interceptors.
+Please find well-argued explanation in the xref:cdi-reference.adoc#repeatable-interceptor-bindings[Repeatable interceptor bindings] section of the CDI reference.
+
+On CDI beans, it is also possible to take custom `java.security.Permission` much further.
+Let's image that in your application, you have following service:
+
+[source,java]
+----
+import io.quarkus.security.PermissionsAllowed;
+import io.smallrye.mutiny.Uni;
+import jakarta.enterprise.context.ApplicationScoped;
+
+import java.security.Permission;
+import java.util.Objects;
+
+@ApplicationScoped
+public class GreetingsService {
+
+    @PermissionsAllowed(value = "ignored", permission = HelloPermission.class) <1>
+    public Uni<String> greetings(int index, Object obj, String hello) {
+        return Uni.createFrom().item("Welcome!");
+    }
+
+    @PermissionsAllowed(value = "ignored", permission = HelloPermission.class, params = "greetings") <2>
```

Comment:
Same question as above

Reply:
This method differs from above in one things - there are 2 strings, what if you want to pass a String to the Permission constructor and it should be just a second one? So you use parameter name to hint Quarkus which parameter should be passed.